docs/reference/setup.asciidoc

```diff
@@ -28,6 +28,10 @@ https://openjdk.java.net[OpenJDK] from the JDK maintainers (GPLv2+CE)
 within each distribution. The bundled JVM is the recommended JVM and
 is located within the `jdk` directory of the Elasticsearch home directory.
 
+WARNING: The bundled JVM requires glibc 2.14+. CentoOS 6 and Oracle Enterprise Linux 6
+have an outdated version of glibc, so you must use your own version of Java
```

Comment:
Should we clarify that you must use a _compatible_ version of Java in those scenarios? Is there some JDK/Centos open issue we can link folks to for more background on this problem?

Reply:
Compatible is a tricky word. Our compatibility matrix doesn't distinguish jdk support crossed by OS, so pointing them there won't help. And I don't know of any upstream issue we can reference. Would it help to clarify Java 14 or lower?

Reply:
Yeah, I'd probably mention Java 14 or lower. That at least gives folks a better idea of what to do so they don't just go setting JAVA_HOME to a JDK 15 install and seeing the same problem.